src/github.com/matrix-org/dendrite/appservice/storage/txn_id_counter_table.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2018 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"context"
+	"database/sql"
+)
+
+const txnIDSchema = `
+-- Keeps a count of the current transaction ID per application service 
+CREATE TABLE IF NOT EXISTS txn_id_counter (
+	-- The ID of the application service the this count belongs to
+	as_id TEXT NOT NULL PRIMARY KEY,
+	-- The last-used transaction ID
+	txn_id INTEGER NOT NULL
+);
```

Comment:
You can use https://www.postgresql.org/docs/9.6/static/sql-createsequence.html, and have a transaction ID be globally unique rather than scoped to an AS.

Reply:
Is there any reason not to have it scoped to an AS though? I did so on purpose so that each AS would get an sequentially incrementing number for transaction ID even if other AS's are working.